airflow/providers/google/cloud/transfers/gcs_to_gcs.py

```diff
@@ -417,6 +420,12 @@ def _copy_directory(self, hook, source_objects, prefix):
                 hook=hook, source_object=source_obj, destination_object=destination_object
             )
 
+    def _check_exact_match(self, source_object: str, prefix: str) -> bool:
+        """Checks whether source_object's name matches the prefix according to the exact_match flag."""
+        if self.exact_match and (source_object != prefix or not source_object.endswith(prefix)):
```

Comment:
I think we can rewrite it to be like 
```suggestion
       return not self.exact_match or (source_object == prefix or source_object.endswith(prefix))
```
But i am not sure if it worth it or not..

Reply:
I moved this condition to a separate method for reuse. I agree it looks confusing, and I honestly don't understand it confidently. That's why I just left is as it is with intention to not break something else.

Reply:
@VladaZakharova , maybe the expression should be like that?
```python
return not self.exact_match or (source_object == prefix and source_object.endswith(prefix))
```

not (A or B) = (not A) and (not B)

Reply:
Then lets use your implementation :)
I agree that this statement looks weird any way :D